packages/language/examples/src/vesting.ts

```diff
@@ -0,0 +1,383 @@
+import {
+  Contract,
+  Case,
+  datetoTimeout,
+  timeoutToDate,
+  close,
+  TokenValue,
+  Timeout,
+  Party,
+  MarloweState,
+  Environment,
+  mkEnvironment,
+  Input,
+} from "@marlowe.io/language-core-v1";
+import {
+  Choice,
+  Deposit,
+  Next,
+  emptyApplicables,
+  noNext,
+} from "@marlowe.io/language-core-v1/next";
+import addMinutes from "date-fns/addMinutes";
+import subMinutes from "date-fns/subMinutes";
+import * as O from "fp-ts/lib/Option.js";
+import { pipe } from "fp-ts/lib/function.js";
+
+/**
+ * Vesting Contract Example
+ * @description : a Token `Provider` block funds on a Marlowe Contract and allow a Token `Claimer` to retrieve them
+ * based on a Vesting Scheme.
+ * Cancel Policy :
+ * 1) At any Vesting Period, The Token Provider can cancel the vesting schedule, then all the downstream vested period will be canceled too.
+ * 2) Once a vested period is claimable by the Claimer, The Provider can't cancel the transaction.
+ * @param request Request For Creating a Vesting Marlowe Contract
+ */
+export const mkContract = function (request: VestingRequest): Contract {
+  const {
+    scheme: { numberOfPeriods },
+  } = request;
+  if (numberOfPeriods < 1)
+    throw "The number of periods needs to be greater or equal to 1";
+
+  return initialEmployerDeposit(request, employeeDepositDistribution(request));
+};
+
+/**
+ * Request For Creating a Vesting Marlowe Contract
+ */
+export type VestingRequest = {
+  /**
+   * The token and its amount to be vested by the provider
+   */
+  tokenValue: TokenValue;
+  /**
+   * The party definition of the Token Provider (Role token or a Cardano Address)
+   */
+  provider: Party;
+  /**
+   * The party definition of the Token Claimer (Role token or a Cardano Address)
+   */
+  claimer: Party;
+  /**
+   * The vesting scheme definition between Token Claimer & Provider
+   */
+  scheme: VestingScheme;
+};
+
+/**
+ * Frequency at which chunks of tokens will be released
+ */
+export type Frequency =
+  | "annually"
+  | "half-yearly"
+  | "quarterly"
+  | "monthly"
+  | "daily"
+  | "hourly";
+
+/**
+ * Vesting Scheme Definition
+ */
+export type VestingScheme = {
+  /**
+   * Start of the vesting schedule
+   */
+  start: Date;
+  /**
+   * Frequency at which chunks of tokens will be released
+   */
+  frequency: Frequency;
+  /**
+   * Number of Periods the Provider wants, to release the totality of the tokens to the claimer.
+   */
+  numberOfPeriods: bigint;
+};
+
+export type State = // Initial Deposit
+
+    | {
+        name: "WaitingDepositByProvider";
+        initialDepositDeadline: Date;
+        depositInput?: Input[];
```

Comment:
What is depositInput in this context? why is it optional and why an array of Inputs?
I would expect a single Input for the initial deposit